doc/feature/compartmentReport.md

```diff
@@ -0,0 +1,422 @@
+         * when read is called the result is undefined.
+         */
+        boost::future< floatsPtr > readNextFrame( );
+
+        /**
+         * Forward to a given absolute timestamp.
+         *
+         * Preconditions:
+         * - r.getState() is OK or ENDED.
+         * - The report was open in read mode.
+         * - There is no previous read or seek operation with a pending future.
+         * - toTimestamp >= getStartTime()
+         *
+         * Postconditions:
+         * Let:
+         *  - r be the CompartmentReport
+         *  - f be the returned future by r.seek(toTimestamp)
+         *  After f.wait() returns:
+         *  - r.getCurrentTime() ~= toTimestamp rounded to the nearest multiple
+              of the timestep, ...
+         *  - ... getState() == OK -> r.getEndTime() > getCurrentTime() and ...
+         *  - ... getState() == ENDED -> r.getEndTime() == getCurrentTime()
+         *
+         * After successful f.wait_for or f.wait_until, the result is the same
+         * as above. If they time out, no observable state changes.
+         *
+         * @throw std::runtime_error if a precondition does not hold.
+         */
+        boost::future< void > seek( const float toTimestamp );
+
+        /** Write the header information of this report.
+         *
+         * Preconditions:
+         * - The report was open in write mode.
+         * - The header has not already been written.
+         *
+         * @param startTime the start time of the report
+         * @param endTime the end time of the report
+         * @param timestep the timestep between report frames
+         * @param dunit the unit of the data, e.g. mV
+         * @param tunit the unit of the time, e.g. ms
+         * @throw std::invalid_argument if any passed argument is invalid
+         */
+        void writeHeader( const float startTime, const float endTime,
+                          const float timestep, const std::string& dunit,
+                          const std::string& tunit );
+
+        /** Write the compartment counts for each section for one cell.
+         *
+         * Preconditions:
+         * - The report was open in write mode.
+         * - writeHeader has already been called.
+         *
+         * @param gid the GID of the cell
+         * @param counts the number of compartments per section
+         * @throw std::runtime_error if writing was not successful.
+         */
+        void writeMapping( const uint32_t gid,
+                           const uint16_ts& counts );
+
+        /** Write the compartment counts for each section for one cell.
+         *
+         * Preconditions:
+         * - The report was open in write mode.
+         * - writeMapping has already been called.
+         *
+         * @param gid the GID of the cell
+         * @param voltages the voltages per compartment to save
+         * @throw std::runtime_error if writing was not successful.
```

Comment:
Can you give an example of when you would like to really write that code. For me a write operation is something that either succeeds or fails, you can use either a return value or a exception to indicate a failure, but I don't find a good reason for using both.